### PR TITLE
Add collection modifying

### PR DESCRIPTION
**Pull request by arn-the-long-beard**
Hello ! Let's implement from the[ Arangodb documentation -> collection modifying ](https://www.arangodb.com/docs/devel/http/collection-modifying.html)

NB : In this todo list, the unit test goes together with the implementation. So an implementation is considered done only if the unit test is also done :). 

- [x] load collection
- [x] unload collection 
- [x] load indexes into memory
- [x] change properties of a collection
- [x] rename collection
- [x] rotate journal
- [x] recalculate count of a collection

Let's go !

### CODE REVIEW

**Comment by arn-the-long-beard** [comment]
I just have seen that I did not use `CollectionType` for  type ( r#type) that we have on our structures.
Should I update them to have Collectiontype instead of u16 ?

**Comment by fMeow** [comment]
Yes. You should use `CollectionType`.

**Comment by arn-the-long-beard** [comment]
Okay ! Thank you for your feedback !

**Comment by arn-the-long-beard** [comment]
I updated with collection type :)
What do you think about collection::load implementation ?

File: src/collection.rs [hunk 1]
```diff
@@ -308,13 +311,25 @@ impl<'a, C: ClientExt> Collection<'a, C> {
...
+    /// The request body object might optionally contain the following attribute:
+    /// - count: If set, this controls whether the return value should include the number of documents in the collection.
+    /// Setting count to false may speed up loading a collection. The default value for count is true.
     /// # Note
     /// this function would make a request to arango server.
     #[maybe_async]
-    pub async fn load(&self) {
-        unimplemented!()
+    pub async fn load(&self, count: bool) -> Result<CollectionLoad, ClientError> {
+        let url = self.base_url.join(&format!("load")).unwrap();
```
Comment (fMeow) [inline src/collection.rs:324]: Here `Url::join` accepts a `&str`, and `"load"` is already a `&str`. So we don't need to allocate memory to store "load" as a `String` and convert it to `&str`.

The following code suffices:
```rust
let url = self.base_url.join("load").unwrap();
```

**Comment by fMeow** [comment]
> I updated with collection type :)
> What do you think about collection::load implementation ?

Quite good I think.

File: src/collection.rs (hunk 1 above)
Comment (arn-the-long-beard) [reply src/collection.rs:324]: Ho yes ! True ! Thank you, I' ll make the update.

**Comment by arn-the-long-beard** [comment]
Ok,  cool !  The json stuff from serde is just awesome !

**Comment by arn-the-long-beard** [comment]
1- There are differences between ArangoDB 3.6 and 3.7 for update properties. Should I implement the [schema validation](https://www.arangodb.com/docs/3.7/data-modeling-documents-schema-validation.html) used in 3.7 ?  

2 - I did not implemented the journalSize because the official documentation states the following :

> APIs that return collection properties or figures will return slightly different attributes for the RocksDB engine than for the MMFiles engine. For example, the attributes journalSize, doCompact, indexBuckets and isVolatile are present in the MMFiles engine but not in the RocksDB engine.

> The MMFiles storage engine is deprecated starting with version 3.6.0 and it will be removed in a future release. To change your MMFiles storage engine deployment to RocksDB, see: Switch storage engine

**Comment by arn-the-long-beard** [comment]
The rename seems ok I think, but I feel some of structures to be redundant. Do you think it is fine to have structs that are pretty similar ? I know it is good in the way that the type is not a giant mess of Option<T> used everywhere but several small structure. But in the other side, we do repeat a lot of code.

**Comment by fMeow** [comment]
> 1 - The rename seems ok I think, but I feel some of structures to be redundant. Do you think it is fine to have structs that are pretty similar ? I know it is good in the way that the type is not a giant mess of Option used everywhere but several small structures. But in the other side, we do repeat a lot of code.
> 
> 2 - It seems the [Recalculate count of a collection](https://www.arangodb.com/docs/stable/http/collection-modifying.html#recalculate-count-of-a-collection) has not been written in collection.rs.
> Is it ok if I add it and implement it? :)

1. I perfer to find a way to abstract over similar structures. We can amass related fields in a `struct` and mark the struct as `Option`.  But I'm not sure if it is a good idea.
2. Sure. Thank you.

**Comment by fMeow** [comment]
> 1- There are differences between ArangoDB 3.6 and 3.7 for update properties. Should I implement the [schema validation](https://www.arangodb.com/docs/3.7/data-modeling-documents-schema-validation.html) used in 3.7 ?
> 
> 2 - I did not implemented the journalSize because the official documentation states the following :
> 
> > APIs that return collection properties or figures will return slightly different attributes for the RocksDB engine than for the MMFiles engine. For example, the attributes journalSize, doCompact, indexBuckets and isVolatile are present in the MMFiles engine but not in the RocksDB engine.
> 
> > The MMFiles storage engine is deprecated starting with version 3.6.0 and it will be removed in a future release. To change your MMFiles storage engine deployment to RocksDB, see: Switch storage engine

1. You can implement this if you like. I don't know if specifying `schema`  using version below 3.7 can cause error. I think ArangoDB server should just ignore this additional and _unknown_ field.

2. For breaking changes across versions or engines, I prefer to use a feature gate to switch between. There is a feature gate names [mmfiles](https://github.com/fMeow/arangors/blob/master/Cargo.toml#L38) and another named [rocksdb](https://github.com/fMeow/arangors/blob/master/Cargo.toml#L39). But it's fine to leave it alone until somebody creates an issue on this.

**Comment by arn-the-long-beard** [comment]
Hey :) Thank you for your messages !

1 - Yes ! Nice to find abstraction. I am just not sure yet on how to do it. We can discuss it later on how to do it.
2 - Okay, I will add the recalculation
3 - I do not have ArangoDB 3.7, so I will need to upgrade in order to make the unit test for it. I will think about it, or maybe I will leave a note in the code so someone could make it.

4 - Nice ! Let's say that the core focus now is on the main functionalities and we can work out details later on :)

**Comment by arn-the-long-beard** [comment]
I will make later a PR for feature gate for versions switch and code improvement in collection.

**Comment by arn-the-long-beard** [comment]
Actually, I forgot to implement [Rotate journal](Rotate journal of a collectionPermalink)

**Comment by arn-the-long-beard** [comment]
OMG, I just have understood how feature gate stuff works. It allows conditional compilation. That is freaking awesome and ergonomic. Rust rocks so hard lol.

File: tests/collection.rs [hunk 2]
```diff
@@ -333,9 +327,321 @@ async fn test_get_checksum() {
...
     assert_eq!(updated_result.status, 3);
-    assert_eq!(updated_result.r#type, 2);
+    assert_eq!(updated_result.r#type, CollectionType::Document);
     assert_eq!(updated_result.checksum.is_empty(), false);
 
     let coll = database.drop_collection(collection_name).await;
     assert_eq!(coll.is_err(), false);
 }
+
+#[maybe_async::test(
```
Comment (fMeow) [inline tests/collection.rs:337]: In the latest version of `maybe-async`, the `test` API has changed and is able to specify multiple async test macros.

In short, use the following code:

```rust
#[maybe_async::test(
    any(feature = "reqwest_blocking"),
    async(any(feature = "reqwest_async"), tokio::test),
    async(any(feature = "surf_async"), async_std::test)
)]
async fn test_put_load() {
// ...
}
```

**Comment by fMeow** [comment]
Fine, I will do the migration works myself.

**Comment by fMeow** [comment]
> OMG, I just have understood how feature gate stuff works. It allows conditional compilation. That is freaking awesome and ergonomic. Rust rocks so hard lol.
> 
> We need somebody to try on mmfiles.

That's why I use docker to set up ArangoDB in e2e test. With docker, we can quick setup an ArangoDB server with any verison and any configuration.

It's fine to just wait for someone to point out that this code errs. But if we want to make for sure before anyone jumps out, we can allocate a new job in CI and setup ArangoDB to `mmfiles`. I migrate CI to github workflow today, but that's as simple as travis CI. I regard it as a description about a pipeline of shell commands.

Maybe I should consider giving up docker compose, and use plain docker command instead.

**Comment by fMeow** [comment]
I have set up a job in [github workflow](https://github.com/fMeow/arangors/runs/803132915?check_suite_focus=true) using ArangoDB server with `mmfiles` engine. Let's see if the rotate_journal works.

**Comment by arn-the-long-beard** [comment]
Nice ! I used to play with Docker some time ago, so not sure . Need to try it out again before I can help with it.
Cool ! having test with ci. Hum it seems the unit test failed for some reason.

**Comment by fMeow** [comment]
Yes, I am working on this.

**Comment by fMeow** [comment]
It seem that collection details in mmfiles and rocksdb are quite different. 

See the latest `develop` branch, I have merge similar fileds on collection into `CollectionDetail` and `CollectionInfo`.

**Pull request merged**